packages/synapse-react-client/src/components/ChallengeRequirementsModal/ChallengeRequirementsModal.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useCallback } from 'react'
+import { ConfirmationDialog } from '../ConfirmationDialog'
+import {
+  useGetAccessRequirementsForTeam,
+  useGetAccessRequirementStatuses,
+  useGetCurrentUserProfile,
+  useGetEntity,
+  useGetEntityChallenge,
+} from '../../synapse-queries'
+import {
+  useAddMemberToTeam,
+  useGetMembershipStatus,
+} from '../../synapse-queries/team/useTeamMembers'
+import { useSynapseContext } from '../../utils'
+import AccessRequirementList from '../AccessRequirementList/AccessRequirementList'
+import { RestrictableObjectType } from '@sage-bionetworks/synapse-types'
+
+export type ChallengeRequirementsModalProps = {
+  open: boolean
+  projectId: string
+  onCancel: () => void
+  onRegisterComplete: () => void
+}
+
+/**
+ * The challenge requirements modal will
+ *  1. Display the access requirements that must be accepted to join a participant team. The component will guide the user through meeting the requirements.
+ *  2. If the user has accepted the requirements, the user can click 'Register' to join the team
+ *     If the user has not accepted the requirements, the user cannot register for the challenge.
+ * @constructor
+ */
+export default function ChallengeRequirementsModal(
+  props: ChallengeRequirementsModalProps,
+) {
+  const { open, projectId, onRegisterComplete, onCancel } = props
+  const { data: project } = useGetEntity(projectId)
+
+  const { accessToken } = useSynapseContext()
+
+  const { data: challenge } = useGetEntityChallenge(projectId)
+  const isLoggedIn = !!accessToken
+  const { data: userProfile } = useGetCurrentUserProfile({
+    enabled: isLoggedIn,
+  })
+
+  const participantTeamId = challenge?.participantTeamId
+  const userId = userProfile?.ownerId
+
+  // Verify that user is a member of the participant team
+  const { data: teamMembershipStatus } = useGetMembershipStatus(
+    participantTeamId!,
+    userId!,
+    {
+      enabled: Boolean(participantTeamId && userId),
+    },
+  )
+
+  const { mutateAsync: addTeamMember } = useAddMemberToTeam()
+
+  const {
+    data: fetchedRequirementsForTeam,
+    isSuccess: hasFetchedRequirements,
+  } = useGetAccessRequirementsForTeam(challenge?.participantTeamId!, {
+    enabled: Boolean(challenge?.participantTeamId),
+  })
+
+  const requirementStatusesQueryResults = useGetAccessRequirementStatuses(
+    fetchedRequirementsForTeam?.map(ar => String(ar.id)) ?? [],
+  )
+
+  const meetsAllRequirements =
+    hasFetchedRequirements &&
+    requirementStatusesQueryResults.every(queryResult => {
+      return queryResult.status === 'success' && queryResult.data.isApproved
+    })
+
```

Comment:
Use this boolean to block attempting to join the team until all access requirements have been met